AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -10,39 +10,32 @@
 
 #import "ASDimension.h"
 #import "ASAssert.h"
+#import "CGRect+ASConvenience.h"
 
 #pragma mark - ASDimension
 
 ASDimension const ASDimensionAuto = {ASDimensionUnitAuto, 0};
 
 ASOVERLOADABLE ASDimension ASDimensionMake(NSString *dimension)
 {
-  // Handle empty string
-  if (dimension.length == 0) {
-    return ASDimensionMake(ASDimensionUnitPoints, 0.0);
-  }
-  
-  // Handle points
-  NSUInteger pointsStringLocation = [dimension rangeOfString:@"pt"].location;
-  if (pointsStringLocation != NSNotFound) {
-    // Check if points is at the end and remove it
-    if (pointsStringLocation == (dimension.length-2)) {
-      dimension = [dimension substringToIndex:(dimension.length-2)];
-      return ASDimensionMake(ASDimensionUnitPoints, dimension.floatValue);
+  if (dimension.length > 0) {
+    
+    // Handle points
+    if ([dimension hasSuffix:@"pt"]) {
+      return ASDimensionMake(ASDimensionUnitPoints, ASCGFloatFromString(dimension));
+    }
+    
+    // Handle auto
+    if ([dimension isEqualToString:@"auto"]) {
```

Comment:
@maicki very minor change, but I would suggest:

- Make auto the last one to check.  It will be the rarest to use since there is an easy constant value, ASDimensionAuto, that feels cleaner for clients to use.
- Check percent first, probably the most common reason to use this?  pt vs percent first doesn't make much difference.

Then, this will be a bit more optimized for anyone who uses it heavily.